playbooks/adhoc/preflight/README.md

```diff
@@ -0,0 +1,43 @@
+# Preflight checks
+
+Here we provide an Ansible playbook for detecting potential roadblocks prior to
+an install or upgrade.
+
+Ansible's default operation mode is to fail fast, on the first error. However,
+when performing checks, it is useful to gather as much information about
+problems as possible in a single run.
+
+The `check.yml` playbook runs a battery of checks against the inventory hosts
+and tells Ansible to ignore intermediate errors, thus giving a more complete
+diagnostic of the state of each host. Still, if any check failed, the playbook
+run will be marked as having failed.
+
+To facilitate understanding the problems that were encountered, we provide a
+custom callback plugin to summarize execution errors at the end of a playbook
+run.
+
+---
+
+*Note that currently the `check.yml` playbook is only useful for RPM-based
+installations. Containerized installs are excluded from checks for now, but
+might be included in the future if there is demand for that.*
+
+---
+
+## Running
+
+With an installation of Ansible 2.2 or greater, run the playbook directly
+against your inventory file. Here is the step-by-step:
+
+1. If you haven't done it yet, clone this repository:
+
+    ```console
+    $ git clone https://github.com/openshift/openshift-ansible
+    $ cd openshift-ansible
+    ```
+
+2. Run the playbook:
+
+    ```console
+    $ ansible-playbook -i <inventory file> playbooks/adhoc/preflight/check.yml
+    ```
```

Comment:
>Ideally the checks should be equally valid for a number of installation paths, not limited to BYO.

>Among the choices (in /playbooks): adhoc, aws, byo, common, gce, libvirt, openstack; I thought adhoc was appropriate for its name not implying any specific target infrastructure, and also that you can run the checks independent of doing an installation.

adhoc was started for tasks that really don't have a home, existing playbooks living there are generally unsupported and not maintained.

common was meant as a generic set of playbooks that would run against different "providers" when we were supporting the bin/cluster tooling. playbooks in common use generic group names and do not line up with the actual groups used by the byo playbooks or bin/cluster derived playbooks.

aws, gce, libvirt, openstack playbook directories are meant to be driven by  bin/cluster, which is community supported and most use is considered deprecated.

>Definitely common would be an option as well, but it's subdirectory structure and the many include: ../common/.../...yml references led me to think common plays a different role: being included in other playbooks for code reuse.

Yes, if the playbook were to live in common, it would be necessary to provide a byo playbook that included the common playbook with the correct group mappings, etc.

>Could we document in the repo what each thing under /playbooks is meant for? I volunteer to create a PR for that if I can get an initial explanation :)

Definitely something that is needed. Let me know if you need any additional detail/description.